### PR TITLE
mask3d functionality updates

### PR DESCRIPTION
**Pull request by GoldenAnpu**
**project_api**
 - `def append_classes` - directly add new classes into the Supervisely project
----
**volume_annotation_api**
 - `def append_objects` - directly add new objects into the Supervisely project
----
**volume_figure_api**
 - `def download_stl_meshes` - fix example in docstring
 - `def append_bulk` - changed some vars but not logic
 - `def _append_bulk_mask3d` - updated some vars and added autoupload for geometries
 - `def upload_sf_geometries` - receives geometries as Dict[UUID, bytes] to conviniently work in def _append_bulk_mask3d
----
**mask_3d**
 - `def create_from_file` - updated Mask3D geometry from file
 - `def from_bytes` - creates Mask3D geometry from bytes (NRRD file represented as bytes)
 -  `def from_json` - now creates geometry with data astype np.bool_
 - `def add_mask_2d` - new method that helps to convert 2d masks to 3d
----
**volume_annotation**
 - `def add_objects` - directly add new objects  to VolumeAnnotation object.
----
**volume_figure**
 - change `def __init__` to accept creation Mask3D (and ClosedSurfaceMesh) without passing arguments plane_name=None and slice_index=None, add types for def init.
----
**volume_object**
 - `def __init__` - add new attribute self.figure if argument mask_3d passed. mask_3d could be file_path, ndarray or Mask3D object

### CODE REVIEW

File: supervisely/api/project_api.py [hunk 1]
```diff
@@ -1202,3 +1204,32 @@ def check_imageset_backup(self, id: int) -> Optional[Dict]:
...
+            os.environ['API_TOKEN'] = 'Your Supervisely API Token'
+            api = sly.Api.from_env()
+
+            proj_id = 28145
+            lung_obj_class = sly.ObjClass("lung", sly.Mask3D)
+            api.project.append_classes(proj_id, [lung_obj_class])
+        """
+        meta_json = self.get_meta(id)
+        meta = ProjectMeta.from_json(meta_json)
+        meta = meta.add_obj_classes(classes)
```
Comment (mkolomeychenko) [inline supervisely/api/project_api.py:1234]: test? ObjClassCollection

File: supervisely/api/volume/volume_figure_api.py [hunk 2]
```diff
@@ -434,22 +438,22 @@ def _append_bulk_mask3d(
...
 
-        fake_figures = []
-        for figure in figures_json:
-            fake_figures.append(
+        empty_figures = []
+        for figure in figures:
+            empty_figures.append(
                 {
-                    "objectId": figure["objectId"],
+                    "objectId": figure.volume_object.to_json(key_id_map)["id"],
```
Comment (mkolomeychenko) [inline supervisely/api/volume/volume_figure_api.py:449]: key_id_map.get_object_id(figure.volume_object.key())

File: supervisely/api/volume/volume_figure_api.py [hunk 3]
```diff
@@ -459,6 +463,70 @@ def _append_bulk_mask3d(
                 figure_id = resp_obj[ApiField.ID]
                 key_id_map.add_figure(key, figure_id)
 
+                for figure in figures:
+                    if figure.key() == key:
+                        geometry = figure.geometry.data
+                        geometry_bytes = encode(geometry.astype(uint8))
+
```
Comment (mkolomeychenko) [inline supervisely/api/volume/volume_figure_api.py:470]: -

Comment (mkolomeychenko) [inline supervisely/api/volume/volume_figure_api.py:470]: -

File: supervisely/geometry/mask_3d.py [hunk 4]
```diff
@@ -483,3 +522,49 @@ def base64_2_data(encoded_string: str) -> np.ndarray:
...
+        """
+
+        from supervisely.volume_annotation.plane import Plane
+
+        Plane.validate_name(plane_name)
+
+        mask_2d = np.fliplr(mask_2d)
+        mask_2d = np.rot90(mask_2d, 1, (1, 0))
+
+        if plane_name == "axial":
```
Comment (mkolomeychenko) [inline supervisely/geometry/mask_3d.py:553]: Plane.AXIAL

Comment (mkolomeychenko) [reply supervisely/geometry/mask_3d.py:553]: many times

File: supervisely/geometry/mask_3d.py [hunk 5]
```diff
@@ -483,3 +522,49 @@ def base64_2_data(encoded_string: str) -> np.ndarray:
...
+            new_shape = self.data.shape[1:]
+        elif plane_name == "coronal":
+            new_shape = self.data.shape[::2]
+
+        if origin:
+            x, y = origin
+            new_mask = np.zeros(new_shape, dtype=mask_2d.dtype)
+            new_mask[x : x + mask_2d.shape[0], y : y + mask_2d.shape[1]] = mask_2d
+
+        if plane_name == "axial":
```
Comment (mkolomeychenko) [inline supervisely/geometry/mask_3d.py:565]: Plane.AXIAL

File: supervisely/geometry/mask_3d.py [hunk 6]
```diff
@@ -483,3 +522,49 @@ def base64_2_data(encoded_string: str) -> np.ndarray:
             data = data.reshape(shape)
             logger.debug("Converted successfully!")
         return data
+
+    def add_mask_2d(
+        self,
+        mask_2d: np.ndarray,
+        plane_name: Literal["axial", "sagittal", "coronal"],
```
Comment (mkolomeychenko) [inline supervisely/geometry/mask_3d.py:529]: literal with Plane.AXIAL

File: supervisely/volume_annotation/volume_figure.py [hunk 7]
```diff
@@ -81,16 +86,29 @@ class VolumeFigure(VideoFigure):
...
-        plane_name,
-        slice_index,
-        key=None,
-        class_id=None,
-        labeler_login=None,
-        updated_at=None,
-        created_at=None,
+        volume_object: VolumeObject,
+        geometry: Geometry,
+        plane_name: Literal["axial", "sagittal", "coronal"] = None,
```
Comment (mkolomeychenko) [inline supervisely/volume_annotation/volume_figure.py:91]: .

File: supervisely/volume_annotation/volume_figure.py [hunk 8]
```diff
@@ -81,16 +86,29 @@ class VolumeFigure(VideoFigure):
...
+        slice_index: int = None,
+        key: Optional[UUID] = None,
+        class_id: Optional[int] = None,
+        labeler_login: Optional[str] = None,
+        updated_at: Optional[str] = None,
+        created_at: Optional[str] = None,
     ):
+        # only Mask3D can be created without 'plane_name' and 'slice_index'
+        if not isinstance(geometry, (Mask3D, ClosedSurfaceMesh)):
+            if plane_name is None and slice_index is None:
```
Comment (mkolomeychenko) [inline supervisely/volume_annotation/volume_figure.py:101]: validate plane name

**Review by mkolomeychenko — CHANGES_REQUESTED** [review]

File: supervisely/api/project_api.py (hunk 1 above)
Comment (GoldenAnpu) [reply supervisely/api/project_api.py:1234]: `ObjClassCollection` works
In `KeyIndexedCollection.add_items()` we have `[*ObjClassCollection]` to extract ObjClass objects in list

File: supervisely/api/volume/volume_figure_api.py (hunk 2 above)
Comment (GoldenAnpu) [reply supervisely/api/volume/volume_figure_api.py:449]: ✅

File: supervisely/volume_annotation/volume_figure.py (hunk 8 above)
Comment (GoldenAnpu) [reply supervisely/volume_annotation/volume_figure.py:101]: already there on 124 line

File: supervisely/geometry/mask_3d.py (hunk 6 above)
Comment (GoldenAnpu) [reply supervisely/geometry/mask_3d.py:529]: circular import if we import `Plane` and Plane.AXIAL can't be used as Litera[Plane.AXIAL]

**Pull request merged**